src/test/java/io/cryostat/recordings/RecordingMetadataManagerTest.java

```diff
@@ -0,0 +1,200 @@
+ *
+ * (a) the Software, and
+ * (b) any piece of software and/or hardware listed in the lrgrwrks.txt file if
+ * one is included with the Software (each a "Larger Work" to which the Software
+ * is contributed by such licensors),
+ *
+ * without restriction, including without limitation the rights to copy, create
+ * derivative works of, display, perform, and distribute the Software and make,
+ * use, sell, offer for sale, import, export, have made, and have sold the
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ * The above copyright notice and either this complete permission notice or at
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package io.cryostat.recordings;
+
+import java.nio.file.OpenOption;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.Map;
+
+import io.cryostat.core.log.Logger;
+import io.cryostat.core.sys.FileSystem;
+
+import com.google.gson.Gson;
+import org.apache.commons.codec.binary.Base32;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.NullSource;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+
+public class RecordingMetadataManagerTest {
+
+    RecordingMetadataManager recordingMetadataManager;
+    @Mock Path recordingMetadataDir;
+    @Mock FileSystem fs;
+    @Mock Base32 base32;
+    @Mock Logger logger;
+
+    @BeforeEach
+    void setup() {
+        Gson gson = new Gson();
+        Base32 base32 = new Base32();
+
+        this.recordingMetadataManager =
+                new RecordingMetadataManager(recordingMetadataDir, fs, gson, base32, logger);
+    }
+
+    @Test
+    void shouldParseAndStoreLabelsInRecordingLabelsMap() throws Exception {
+        String targetId = "someTarget";
+        String recordingName = "someRecording";
+        String labels = "{\"KEY\":\"VALUE\",\"key.2\":\"some.value\",\"key3\":\"1234\"}";
+
+        Path mockPath = Mockito.mock(Path.class);
+        Mockito.when(recordingMetadataDir.resolve(Mockito.anyString())).thenReturn(mockPath);
```

Comment:
I'm getting a null pointer exception at this line when attempting to mock the `fs.writeString()` call. I tried following the rule registry test [here](https://github.com/cryostatio/cryostat/blob/main/src/test/java/io/cryostat/rules/RuleRegistryTest.java#L132) and am not sure what to try next. Any ideas?

```
java.lang.NullPointerException
 at io.cryostat.recordings.RecordingMetadataManagerTest.shouldOverwriteLabelsForExistingLabelEntries([RecordingMetadataManagerTest.java:179]())
 at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
 at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke([NativeMethodAccessorImpl.java:62]())
 at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke([DelegatingMethodAccessorImpl.java:43]())
 at java.base/java.lang.reflect.Method.invoke([Method.java:566]())
 at org.junit.platform.commons.util.ReflectionUtils.invokeMethod([ReflectionUtils.java:688]())
```

Reply:
I think you're just missing the `@ExtendWith` annotation, like here: https://github.com/cryostatio/cryostat/blob/main/src/test/java/io/cryostat/rules/RuleRegistryTest.java#L71

Without that, the `@Mock` on your fields doesn't do anything, so `recordingMetadataDir` is actually an uninitialized null when you get to this line.

Reply:
Thanks!